proxy/request_test.go

```diff
@@ -0,0 +1,25 @@
+package proxy
+
+import (
+	"testing"
+)
+
+func TestBadMessage(t *testing.T) {
+	req := ""
+	resp := handleRequest([]byte(req))
+
+	expected := `{"id":null,"error":{"errcode":"M_NOT_JSON","error":"unexpected end of JSON input"}}`
+	if string(resp) != expected {
+		t.Error("Got response", string(resp), " expected ", expected)
```

Comment:
Typical format is:

t.Error("Bad response: want:", expected, "got:", resp)

Notably, the words "want" and "got", and the order "want" then "got"